core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -248,6 +248,64 @@ void* SYCLInternal::scratch_flags(
   return m_scratchFlags;
 }
 
+template <sycl::usm::alloc Kind>
+size_t SYCLInternal::USMObjectMem<Kind>::reserve(size_t n) {
+  assert(m_size == 0);
+  assert(m_q);
+
+  if (m_capacity < n) {
+    using Space  = std::conditional_t<Kind == sycl::usm::alloc::device,
+                                     Kokkos::Experimental::SYCLDeviceUSMSpace,
+                                     Kokkos::Experimental::SYCLSharedUSMSpace>;
```

Comment:
You are using this pattern in three places.  Did you consider declaring it as a member type?

Reply:
Yes, I thought about it but couldn't decide if it's worth it or not. I can change.